src/Skynet.Protocol/ChannelMessage.cs

```diff
@@ -0,0 +1,151 @@
+﻿using Skynet.Model;
+using Skynet.Network;
+using Skynet.Protocol.Cryptography;
+using Skynet.Protocol.Model;
+using System;
+using System.Collections.Generic;
+
+namespace Skynet.Protocol
+{
+    public class ChannelMessage : Packet
+    {
+        public byte PacketVersion { get; set; }
+        public long ChannelId { get; set; }
+        public long SenderId { get; set; }
+        public long MessageId { get; set; }
+        public long SkipCount { get; set; }
+        public DateTime DispatchTime { get; set; }
+        public MessageFlags MessageFlags { get; set; }
+        public long FileId { get; set; }
+        public ChannelMessageFile File { get; set; }
+        public List<Dependency> Dependencies { get; set; } = new List<Dependency>();
+
+        public ReadOnlyMemory<byte>? PacketContent { get; set; }
+
+        public MessageFlags RequiredFlags { get; set; } = MessageFlags.None;
+        public MessageFlags AllowedFlags { get; set; } = MessageFlags.All;
+
+        public override Packet Create() => new ChannelMessage().Init(this);
+
+        public void Decrypt(ReadOnlySpan<byte> key)
+        {
+            if (key.Length != 64) throw new ArgumentOutOfRangeException(nameof(key), key.Length, "The key must be exactly 64 bytes long.");
+            if (PacketContent == null) throw new InvalidOperationException("You cannot decrypt a message that has not been read");
+
+            byte[] hmacKey = key.Slice(0, 32).ToArray();
+            byte[] aesKey = key.Slice(32, 32).ToArray();
+            ReadContent(AesStatic.EncryptWithHmac(PacketContent.Value, hmacKey, aesKey));
+            Array.Clear(hmacKey, 0, 32);
```

Comment:
Yes, the array is cleared for security reasons. In other places that would actually help more than here as this array is a very short lived object and therefore overridden by the GC early. However, it seems to be a good practice to clear keys and IVs like in [`System.Security.Cryptography.SymmetricAlgorithm`](https://source.dot.net/#System.Security.Cryptography.Primitives/System/Security/Cryptography/SymmetricAlgorithm.cs,b554cc06a34c922e). If you think it's necessary to do this everywhere please create a new issue.

Reply:
I think that if we implement it here, it should be consistently used everywhere, because if the key leaks somewhere else, then the array clear here is kinda unneccessary.

Reply:
See #2